cmd/weaver/main.go

```diff
@@ -74,11 +74,25 @@ func main() {
 	switch flag.Arg(0) {
 	case "generate":
 		generateFlags := flag.NewFlagSet("generate", flag.ExitOnError)
+		tags := generateFlags.String("tags", "not specified", "Optional tags for the generate command")
 		generateFlags.Usage = func() {
 			fmt.Fprintln(os.Stderr, generate.Usage)
 		}
 		generateFlags.Parse(flag.Args()[1:])
-		if err := generate.Generate(".", flag.Args()[1:], generate.Options{}); err != nil {
+		buildTags := "--tags=ignoreWeaverGen"
+		if *tags != "not specified" { // tags flag was specified
+			if *tags == "" || *tags == "." {
+				// User specified the tags flag but didn't provide any tags.
+				fmt.Fprintln(os.Stderr, "No tags provided.")
+				os.Exit(1)
+			}
+			// TODO(rgrandl): we assume that the user specify the tags properly. I.e.,
+			// a single tag, or a list of tags separated by comma. We may want to do
+			// extra validation at some point.
+			buildTags = buildTags + "," + *tags
+		}
+		idx := len(flag.Args()) - len(generateFlags.Args())
```

Comment:
Why do we need to use flag.Args()[idx:] instead passing generateFlags.Args() to generate.Generate?

Reply:
Great point again.